cmd/batch-expire.go

```diff
@@ -427,7 +427,7 @@ func batchObjsForDelete(ctx context.Context, r *BatchJobExpire, ri *batchJobInfo
 					default:
 					}
 					stopFn := globalBatchJobsMetrics.trace(batchJobMetricExpire, ri.JobID, attempts)
-					_, err := api.DeleteObject(ctx, exp.Bucket, exp.Name, ObjectOptions{
+					_, err := api.DeleteObject(ctx, exp.Bucket, encodeDirObject(exp.Name), ObjectOptions{
```

Comment:
I can see this is encoded inside erasureServerPools.DeleteObject() as well, I do not think this is needed

Reply:
In erasureServerPools.DeleteObject, encodeDirObject is called only when opts.DeletePrefix is false. In both these cases opts.DeletePrefix is true.

Reply:
@vadmeste it is not - it is doing extra work see why this is needed from Decommission code.